Refactor Init error window to fit to design spec

jarolrod opened the pull request:
Refactors the init error window to fit to the design spec: [init error window design spec](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App?node-id=4204%3A85964)

| master | PR |
| --------- | ---- |
| <img width="612" alt="Screen Shot 2022-08-15 at 10 18 07 PM" src="https://user-images.githubusercontent.com/23396902/184787821-693e72aa-e4ac-4535-b6cb-b9b3f780e1ce.png"> | <img width="612" alt="Screen Shot 2023-01-16 at 1 53 50 AM" src="https://user-images.githubusercontent.com/23396902/212616368-688fc9b0-def8-442c-b976-e5592ae09881.png"> |

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/164)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/164)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/164)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/164)

closes  bitcoin-core/gui-qml#130

jarolrod commented:
Should be noted that it seems the design calls for a version of inter thats `bold` instead of `semi-bold`

GBKS commented:
Looking at the screenshot, the body text doesn't seem quite right yet (font and font size).

jarolrod commented:
updated from 9862f87 to  6375791 

changes: rebased over changes on dependent PR's

jarolrod commented:
Updated from 6375791 to 989c49b

changes: fixed font issue, thanks @GBKS for spotting that!

hebasto commented:
> Based on #162 and #162

:)

Anyway, could be rebased.

jarolrod commented:
setting as draft until design is settled

jarolrod commented:
Updated from 989c49b to 2d427d9 

changes: rebased over main

jarolrod commented:
updated from from 2d427d9 to 88d9beb

changes: remove irrelevant color definition in dummy rectangle

jarolrod commented:
updated from [88d9beb ](https://github.com/bitcoin-core/gui-qml/commit/88d9bebb059701187a65e24e54f4b4e6244b9bfa)to [c06c93c](https://github.com/bitcoin-core/gui-qml/commit/c06c93c4f3c1215b501004177ccde08663be8261)

Changes: Adjusted to fit the following design: [init-error-window](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App?node-id=4735%3A107912), but not in it's entirety. Follow-ups to complete the design added to PR descirption for anyoe to pick up.

johnny9 commented on src/qml/bitcoin.cpp:
```diff
@@ -67,6 +67,7 @@ bool InitErrorMessageBox(
 {
     QQmlApplicationEngine engine;
     engine.rootContext()->setContextProperty("message", QString::fromStdString(message.translated));
+    engine.addImageProvider(QStringLiteral("images"), new ImageProvider{NULL});
```
Did you mean to add this change to the commit?

jarolrod replied:
thanks for spotting that, this was from an earlier iteration of the PR where we were creating a custom os independent title bar for the window; we're no longer doing that. removed

jarolrod commented:
Updated from c06c93c to 86fde6b

changes:
- rebased over changes on master
- removed unnecessary image provider from the initerrorwindow

johnny9 reviewed (approved):
ACK 86fde6b378adf4c1d456e862abe3fb140ad28ffa

Tested by feeding bitcoin-qt an invalid cli command.
